pandas/core/arrays/datetimes.py

```diff
@@ -376,6 +377,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif is_int64_dtype(dtype):
+            return self.asi8
+
+        # TODO: warn that dtype is not used?
```

Comment:
You can remove this todo I think, as the dtype will still be used by numpy (numpy will ensure after calling this the correct dtype is returned, it seems)

Reply:
Oh good, will do